Srclib.Nuget/ConsoleCommands/DepresolveConsoleCommand.cs

```diff
@@ -10,124 +10,129 @@
+                };
+                // the first library description is always self-reference, so skip it
+                return ApplicationHostContext.GetRuntimeLibraries(context, false).Skip(1);
+            })
+            .Distinct(LibraryUtils.Comparer)
+            .OrderBy(l => l.Identity?.Name);
+
+
+        static async Task RunResolve(string dir)
+        {
+            var p = new Process();
+            p.StartInfo.WorkingDirectory = dir;
+            p.StartInfo.FileName = _dnuPath.Value;
+            p.StartInfo.Arguments = "restore";
+            p.StartInfo.UseShellExecute = false;
+            p.StartInfo.CreateNoWindow = true;
+            p.StartInfo.RedirectStandardInput = true;
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.RedirectStandardError = true;
+            p.Start();
+            // it's important to read stdout and stderr, else it might deadlock
+            var outs = await Task.WhenAll(p.StandardOutput.ReadToEndAsync(), p.StandardError.ReadToEndAsync());
+            p.WaitForExit();
+        }
+
+        static string FindDnuNix()
+        {
+            return RunForResult("/bin/bash", "-c \"which dnu\"");
+        }
 
-    static string FindDnuNix()
-    {
-      return RunForResult("/bin/bash", "-c \"which dnu\"");
-    }
+        public static string RunForResult(string shell, string command)
+        {
+            var p = new Process();
+            p.StartInfo.WorkingDirectory = Directory.GetCurrentDirectory();
+            p.StartInfo.FileName = shell;
+            p.StartInfo.Arguments = command;
+            p.StartInfo.UseShellExecute = false;
+            p.StartInfo.CreateNoWindow = true;
+            p.StartInfo.RedirectStandardInput = true;
+            p.StartInfo.RedirectStandardOutput = true;
+            p.StartInfo.RedirectStandardError = false;
+
+            p.Start();
+            var result = p.StandardOutput.ReadToEnd().Trim();
+            p.WaitForExit();
+            return result;
+        }
+
+        public static string GetDll(string path)
+        {
+            return RunForResult("/bin/bash", "-c \"cd " + path + " && find -name *.dll | head -1\"");
+        }
 
-    static string RunForResult(string shell, string command)
-    {
-      var p = new Process();
-      p.StartInfo.WorkingDirectory = Directory.GetCurrentDirectory();
-      p.StartInfo.FileName = shell;
-      p.StartInfo.Arguments = command;
-      p.StartInfo.UseShellExecute = false;
-      p.StartInfo.CreateNoWindow = true;
-      p.StartInfo.RedirectStandardInput = true;
-      p.StartInfo.RedirectStandardOutput = true;
-      p.StartInfo.RedirectStandardError = false;
-
-      p.Start();
-      var result = p.StandardOutput.ReadToEnd().Trim();
-      p.WaitForExit();
-      return result;
+        public static string GetDll(string path, string name)
+        {
+            return RunForResult("/bin/bash", "-c \"cd " + path + " && find -name " + name + ".dll | head -1\"");
```

Comment:
same here, instead of shelling you, you could do this in code. See https://msdn.microsoft.com/en-us/library/bb513869.aspx